packages/apps/plugins/plugin-sheet/src/components/Grid/Grid.tsx

```diff
@@ -135,25 +137,37 @@ const GridRoot = ({ children, readonly, sheet }: PropsWithChildren<GridContextPr
 // Main
 //
 
-type GridMainProps = Partial<GridBounds>;
+type GridMainProps = { className?: string } & Partial<GridBounds>;
 
-const GridMain = ({ numRows, numColumns }: GridMainProps) => {
+const GridMain = ({ className, numRows, numColumns }: GridMainProps) => {
   const { model, cursor, setCursor, setRange, setEditing } = useGridContext();
 
   // Scrolling.
   const { rowsRef, columnsRef, contentRef } = useScrollHandlers();
 
   //
-  // Row/columns.
-  // TODO(burdon): Listen for changes.
+  // Order of Row/columns.
   //
   const [rows, setRows] = useState([...model.sheet.rows]);
   const [columns, setColumns] = useState([...model.sheet.columns]);
   useEffect(() => {
-    setRows([...model.sheet.rows]);
-    setColumns([...model.sheet.columns]);
+    const rowsAccessor = createDocAccessor(model.sheet, ['rows']);
+    const columnsAccessor = createDocAccessor(model.sheet, ['columns']);
+    const handleUpdate = debounce(() => {
+      setRows([...model.sheet.rows]);
+      setColumns([...model.sheet.columns]);
+    }, 100);
+
+    rowsAccessor.handle.addListener('change', handleUpdate);
```

Comment:
Should setup like this be factored out to the model do you think?

Reply:
i don't think so. i think this is what the model does. the component might define an abstract model and this is the ECHO implementation, but otherwise i think it would be way too complex.